### PR TITLE
LCAM-223|New end point to process crown court actions

### PR DESCRIPTION
**Pull request by venkat980**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/dto/CrownCourtsActionsRequestDTO.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package uk.gov.justice.laa.crime.crowncourt.dto;
+
+public class CrownCourtsActionsRequestDTO {
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/dto/CrownCourtsActionsRequestDTO.java:3]: Can we not flesh this out now that we know the schema requirements?

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/config/CrownCourtProceedingTestConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
+import org.springframework.security.oauth2.server.authorization.client.JdbcRegisteredClientRepository;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClient;
+import org.springframework.security.oauth2.server.authorization.client.RegisteredClientRepository;
+import org.springframework.security.oauth2.server.authorization.config.ClientSettings;
+
+import java.util.UUID;
+
+@Configuration(proxyBeanMethods = false)
+public class CrownCourtProceedingTestConfiguration {
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/config/CrownCourtProceedingTestConfiguration.java:18]: Do we need this right now? The authorisation server config hasn't been added yet

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingController.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/internal/v1/crowncourt/actions")
+@Tag(name = "Crown Court Proceeding", description = "Rest API for Crown Court Proceeding.")
+public class CrownCourtProceedingController {
+
+    private final CrownCourtProceedingService crownCourtProceedingService;
+
+    private CrownCourtsActionsRequestDTO preProcessRequest(ApiCheckCrownCourtActionsRequest crownCourtActionsRequest) {
+        return new CrownCourtsActionsRequestDTO();
```
[the remark this answers is not in the document]
Comment (venkat980) [reply crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownCourtProceedingController.java:35]: Done

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/config/CrownCourtProceedingTestConfiguration.java (hunk 2 above)
Comment (venkat980) [reply crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/config/CrownCourtProceedingTestConfiguration.java:18]: Authorization is now added.

**Review by msahilan07 — APPROVED** [review]

**Pull request merged**